### PR TITLE
TST: better skips in testing

### PR DESCRIPTION
**Pull request by jreback**
skip testing on not-installed impala packages
skip testing sqlite on non-existant testing db
add useful Makefile directives

### CODE REVIEW

File: ibis/sql/postgres/tests/conftest.py [hunk 1]
```diff
@@ -19,9 +19,6 @@
 
 import ibis
 
-pytest.importorskip('sqlalchemy')
-pytest.importorskip('psycopg2')
```
Comment (cpcloud) [inline ibis/sql/postgres/tests/conftest.py]: Why did you remove these?

Comment (jreback) [inline ibis/sql/postgres/tests/conftest.py]: otherwise the w/o psycopg2 installed pytest completely fails to even collect. I don't think you are supposed to explicity skip in a conftest like this.

File: Makefile [hunk 2]
```diff
@@ -8,3 +8,12 @@ impala-test:
...
 	find . -name "*.pyc" -exec rm -rf {} \;
+
+develop: clean-pyc
+	python setup.py develop
+
+lint:
+	flake8
+
+test:
+	pytest --pyargs ibis
```
Comment (cpcloud) [inline Makefile]: this should probably include `-m 'not impala and not hdfs'` because you can have all the python deps necessary to run impala queries but no impala nodes up to test against.

Comment (jreback) [inline Makefile]: sure

File: ibis/sql/sqlite/tests/conftest.py [hunk 3]
```diff
@@ -13,7 +13,10 @@
 def dbpath():
     # If we haven't defined an environment variable with the path of the SQLite
     # database, assume it's in $PWD
-    return os.environ.get('IBIS_TEST_SQLITE_DB_PATH', 'ibis_testing.db')
+    path = os.environ.get('IBIS_TEST_SQLITE_DB_PATH', 'ibis_testing.db')
+    if not os.path.exists(path):
+        pytest.skip("sql testing db does not exist!")
+    return path
```
Comment (cpcloud) [inline ibis/sql/sqlite/tests/conftest.py]: Does `pytest.skip` raise an exception so that this is never reached?

Comment (jreback) [inline ibis/sql/sqlite/tests/conftest.py]: yes, this is similar to how the decorator works, but you can do it *inside* a function.

File: ibis/sql/postgres/tests/conftest.py (hunk 1 above)
Comment (cpcloud) [inline ibis/sql/postgres/tests/conftest.py]: Interesting. Those should be moved to `postgresql/test_functions.py` then. We still need to skip if these aren't installed.

Comment (jreback) [inline ibis/sql/postgres/tests/conftest.py]: yes, they are already there :>

Comment (cpcloud) [inline ibis/sql/postgres/tests/conftest.py]: Oh ha! nice.

File: ibis/sql/sqlite/tests/conftest.py (hunk 3 above)
Comment (cpcloud) [inline ibis/sql/sqlite/tests/conftest.py]: cool

**Comment by cpcloud** [comment]
@jreback If you enable CircleCI and Appveyor on your fork the build will be much faster.

**Comment by jreback** [comment]
> This is failing because you can't import impyla.

> You need to skip on import impala

I think I actually pushed *right* before you wrote this....hahah

File: ibis/impala/tests/common.py [hunk 4]
```diff
@@ -23,6 +23,12 @@
 import ibis.compat as compat
 import ibis
 
+try:
+    import impyla
```
Comment (cpcloud) [inline ibis/impala/tests/common.py]: What you had before using `importorskip` is fine, you just need to change `impyla` to `impala`.

**Comment by jreback** [comment]
pushed

**Comment by cpcloud** [comment]
thanks @jreback!
